### PR TITLE
feat!: replace RegExp with RE2

### PR DESCRIPTION
**Pull request by shuplenkov**
## Issue being fixed or feature implemented

To secure user regex we need to  use re2 module instead of regex in ajv and ajv formats

## What was done?

Replaced RegExp with RE2 in Ajv
Replaced RegExps in DataContract schema with compatible ones
New IncompatibleRe2PatternError for incompatible patterns

## How Has This Been Tested?

Unit tests

## Breaking Changes

- `pattern` keyword accept only RE2 compatible regular expressions https://github.com/google/re2/wiki/Syntax 
- Document type and properties minimum length is 3 chars

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by antouhou — COMMENTED** [review]
Looks good 👍 Now let's wait for the re2-wasm to be published

File: package.json [hunk 1]
```diff
@@ -71,13 +72,16 @@
     "@dashevo/feature-flags-contract": "0.1.0",
     "ajv": "^7.2.4",
     "ajv-formats": "^2.1.0",
+    "bignumber.js": "^9.0.1",
```
Comment (shumkov) [inline package.json:75]: Do we use it?

File: package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
     "test:coverage": "NODE_ENV=test nyc --check-coverage --stmts=98 --branch=94 --funcs=95 --lines=98 mocha 'test/unit/**/*.spec.js' 'test/integration/**/*.spec.js'",
     "prepublishOnly": "npm run build"
   },
+  "browser": "dist/DashPlatformProtocol.min.js",
```
Comment (shumkov) [inline package.json:14]: Don't forget to remove

File: package.json (hunk 1 above)
Comment (shuplenkov) [reply package.json:76]: There is a warning from CBOR, that it needs it.

Comment (shumkov) [reply package.json:75]: I got it thanks

File: lib/util/getRE2Class.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+const EventEmitter = require('events');
```
Comment (shumkov) [inline lib/util/getRE2Class.js:1]: Maybe, it’s better to move it to the re2-wasm repo?

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]
Good job!

**Pull request merged**